nested/mssql.json

```diff
@@ -0,0 +1,63 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01-preview/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "moodleCommon": {
+            "metadata": {
+                "description": "Common Moodle values"
+            },
+            "type": "object"
+        }
+    },
+    "resources": [
+        {
+            "apiVersion": "2015-05-01-preview",
```

Comment:
Is there any special reason you are using a preview apiVersion? If not needed, I think we should stick with a non-preview version, and I see "2014-04-01" is the published apiVersion for Microsoft.Sql/servers at [here](https://docs.microsoft.com/en-us/rest/api/sql/servers/createorupdate).

Reply:
It fails to build a stack when using "2014-04-01" in place of "2015-05-01-preview".  The error output is:

> 
Deployment failed. Correlation ID: 5ab06209-bc2e-407e-93e3-467c89e5d0e7. {
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "DeploymentFailed",
        "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.",
        "details": [
          {
            "code": "BadRequest",
            "message": "{\r\n  \"code\": \"BadRequest\",\r\n  \"message\": \"An error occurred while processing this request.\",\r\n  \"target\": null,\r\n  \"details\": [],\r\n  \"innererror\": []\r\n}"
          }
        ]
      }
    ]
  }
}

>

Reply:
I see. Too bad that the error message is almost useless. I guess we'll have to stick with the original apiVersion. Sorry about that... I guess we could try debugging this, but that may not be appropriate on this PR. Let me try that later after this PR is merged.